t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -680,9 +727,9 @@ test_expect_success 'sparse-index is not expanded' '
 	ensure_not_expanded checkout - &&
 	ensure_not_expanded switch rename-out-to-out &&
 	ensure_not_expanded switch - &&
-	git -C sparse-index reset --hard &&
+	ensure_not_expanded reset --hard &&
 	ensure_not_expanded checkout rename-out-to-out -- deep/deeper1 &&
-	git -C sparse-index reset --hard &&
+	ensure_not_expanded reset --hard &&
```

Comment:
These changes could be included in the commit that allows `git reset --hard` to work with the sparse index, to demonstrate that the code change is the reason this test change makes sense.

Reply:
Updated! Also, noted that tests should generally be embedded in the commits corresponding to their changes (unless the standalone change is "add more tests").